llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -3488,6 +3490,14 @@ void llvm::copyMetadataForLoad(LoadInst &Dest, const LoadInst &Source) {
       copyRangeMetadata(DL, Source, N, Dest);
       break;
     }
+    // Extended last-use / nontemporal hint on AMD GPUs
+    if (ID == Ctx.getMDKindID("amdpu.last.use"))
+      Dest.setMetadata(ID, N);
+    // Currently only relevant to atomics
+    else if (ID == Ctx.getMDKindID("amdgpu.no.remote.memory"))
+      Dest.setMetadata(ID, N);
+    else if (ID == Ctx.getMDKindID("amdgpu.no.fine.grained.memory"))
+      Dest.setMetadata(ID, N);
```

Comment:
These cases aren't relevant to loads, so no reason to clutter up generic code with these. Or we should promote these to recognized, enum metadata

Reply:
So, while they're currently only related to `atomicrmw`s, I can imagine some hypothetical future where an atomic load or the like needs a `!amdgpu.no.host.memory` to avoid setting some pessimizing flag

Reply:
This is too ugly to speculatively save here. If we want to move this, we should move them to be first class recognized enum metadata

Reply:
I think I see your point re the atomic hints, but the `!amdgpu.last.use` would want to be an expanded `nontemporal`